reco_utils/dataset/criteo_dac.py

```diff
@@ -1,73 +1,164 @@
+
+
+CRITEO_URL_FULL = "https://s3-eu-west-1.amazonaws.com/kaggle-display-advertising-challenge-dataset/dac.tar.gz"
+CRITEO_URL_SAMPLE = (
+    "http://labs.criteo.com/wp-content/uploads/2015/04/dac_sample.tar.gz"
+)
+CRITEO_URL = {"full": CRITEO_URL_FULL, "sample": CRITEO_URL_SAMPLE}
+DEFAULT_HEADER = ["label"] + ["int{0:02d}".format(i) for i in range(13)] + ["cat{0:02d}".format(i) for i in range(26)]
+
 
 def load_pandas_df(
-        local_cache_path="dac_sample.tar.gz",
-        label_col = 'Label',
-        nume_cols = ['I'+str(i) for i in range(1, 14)],
-        cate_cols = ['C'+str(i) for i in range(1, 27)]
+    size="sample",
+    local_cache_path="dac_sample.tar.gz",
+    header=DEFAULT_HEADER
 ):
-  """Loads the Criteo DAC dataset as pandas.DataFrame.
+    """Loads the Criteo DAC dataset as pandas.DataFrame. This function download, untar, and load the dataset.
 
-  Download the dataset from http://labs.criteo.com/wp-content/uploads/2015/04/dac_sample.tar.gz, untar, and load
+    The schema is:
+    <label> <integer feature 1> ... <integer feature 13> <categorical feature 1> ... <categorical feature 26>
 
-  For the data, the schema is:
-  
-  <label> <integer feature 1> ... <integer feature 13> <categorical feature 1> ... <categorical feature 26>
-
-  Args:
-      local_cache_path (str): Path where to cache the tar.gz file locally
-      label_col (str): The column of Label.
-      nume_cols (list): The names of numerical features.
-      cate_cols (list): The names of categorical features.
-  Returns:
-      pandas.DataFrame: Criteo DAC sample dataset.
-  """
-
-  ## download and untar the data file
-  datapath = _load_datafile(local_cache_path=local_cache_path)
-  df = pd.read_csv(datapath, sep="\t", header=None, names=[label_col]+nume_cols+cate_cols)
-  
-  return df
+    More details: http://labs.criteo.com/2013/12/download-terabyte-click-logs/
 
-def _load_datafile(local_cache_path="dac_sample.tar.gz"):
-    """ Download and extract file """
+    Args:
+        size (str): Dataset size. It can be "sample" or "full".
+        local_cache_path (str): Path where to cache the tar.gz file locally
+        header (list): Dataset header names.
+
+    Returns:
+        pd.DataFrame: Criteo DAC sample dataset.
+    """
+    path, filename = os.path.split(os.path.realpath(local_cache_path))
+    download_criteo(size, filename, path)
+    filepath = extract_criteo(size, path, local_cache_path)
+    return pd.read_csv(filepath, sep="\t", header=None, names=header)
+
+
+def load_spark_df(
+    spark,
+    size="sample",
+    header=DEFAULT_HEADER,
+    local_cache_path="dac.tar.gz",
+    dbfs_datapath="dbfs:/FileStore/dac",
+    dbutils=None,
+):
+    """Loads the Criteo DAC dataset as pySpark.DataFrame.
+
+    The schema is:
+    <label> <integer feature 1> ... <integer feature 13> <categorical feature 1> ... <categorical feature 26>
+
+    More details: http://labs.criteo.com/2013/12/download-terabyte-click-logs/
```

Comment:
wrong reference, this is the display advertising challenge dataset

Reply:
you have to accept the conditions to see the details. I added a note and also give more context in the text